unstructured/metrics/evaluate.py

```diff
@@ -585,3 +604,95 @@ def filter_metrics(
         raise ValueError("Please provide `export_filename`.")
     else:
         raise ValueError("Return type must be either `dataframe` or `file`.")
+
+
+@dataclass
+class ObjectDetectionMetricsCalculator(BaseMetricsCalculator):
+    """
+    Calculates object detection metrics for each document:
+    - f1 score
+    - precision
+    - recall
+    - average precision (mAP)
+    It also calculates aggregated metrics.
+    """
+
+    def __post_init__(self):
+        super().__post_init__()
+        self._document_paths = [
+            path.relative_to(self.documents_dir)
+            for path in self.documents_dir.rglob("analysis/*/layout_dump/object_detection.json")
+        ]
+
+    @property
+    def supported_metric_names(self):
+        return ["f1_score", "precision", "recall", "m_ap"]
+
+    @property
+    def default_tsv_name(self):
+        return "all-docs-object-detection-metrics.tsv"
+
+    @property
+    def default_agg_tsv_name(self):
+        return "aggregate-object-detection-metrics.tsv"
+
+    def _find_file_in_ground_truth(self, file_stem: str):
+        for path in self._ground_truth_paths:
+            if Path(path.stem).stem == file_stem:
```

Comment:
Looks like the intention was simply `path.stem`. If that's not the case I think it may require a comment.

Reply:
This one I care for wrt to approval

Reply:
That's tricky and I think requires a comment because looks misleading.
Generally the ground truth files are json but the original extension is kept.
So for source files:
```
some_document.pdf
poster.jpg
```
ground truth files will look like:
```
some_document.pdf.json
poster.jpg.json
```

So to take real `stem` we need to apply it twice. I'll add a comment for that.

Reply:
Added a descriptive docstring